### ISSUE TITLE
Unify RPC connection creation for chain backends

### ISSUE THREAD

**Issue by yyforyongyu** [lightningnetwork/lnd#7951]
When using `bitcoind` as the chain backend, `lnd` creates three RPC connections,
1. using [`chain.NewBitcoindConn`](https://github.com/lightningnetwork/lnd/blob/13568fd5b09435f8254a91527996091bef30996e/chainreg/chainregistry.go#L435) in `chainreg`.
2. using [`rpcclient.New`](https://github.com/lightningnetwork/lnd/blob/13568fd5b09435f8254a91527996091bef30996e/chainreg/chainregistry.go#L507) in `chainreg`.
3. using [`rpcclient.New`](https://github.com/lightningnetwork/lnd/blob/13568fd5b09435f8254a91527996091bef30996e/lnwallet/chainfee/estimator.go#L321) in `chainfee`.

This could cause issues as `lnd`may make concurrent RPC requests to `bitcoind` which may cause failures. This also breaks the abstraction provided by `chain.BitcoindConn` as we are now accessing the connection directly. To fix it,
- [ ] replace the `rpcclient.New` used in 2 and 3 with the `chain.BitcoindConn` created in 1, to make sure we only ever create one RPC connection to `bitcoind`.
- [ ] replace other occurrences of `rpcclient.New` with `chain.NewRPCClient` for `btcd` backend.
- [ ] rename `chain.NewRPCClient` to `chain.NewBtcdClient`  in `btcwallet ` for clarity.

**Comment by mohamedawnallah** [comment]
Hi @yyforyongyu, 

I'm interested in working on this issue. After taking a closer look, it seems we need to submit two PRs in a sequential order:

### 1️⃣ [btcsuite/btcwallet](https://github.com/btcsuite/btcwallet) Repo
- Add `GetBlockChainInfo` and `RawRequest` APIs to [`Interface`](https://github.com/btcsuite/btcwallet/blob/93c858b2ad6390f574f73dc05d3c610cb78a8efc/chain/interface.go#L34).
- Implement these APIs for [`BitcoindClient`](https://github.com/btcsuite/btcwallet/blob/93c858b2ad6390f574f73dc05d3c610cb78a8efc/chain/bitcoind_client.go#L30), and return a "not implemented" error for [`NeutrinoClient`](https://github.com/btcsuite/btcwallet/blob/93c858b2ad6390f574f73dc05d3c610cb78a8efc/chain/neutrino.go#L29).
- Update [`mockChainClient`](https://github.com/btcsuite/btcwallet/blob/93c858b2ad6390f574f73dc05d3c610cb78a8efc/wallet/mock.go#L14) to mock the implementation of these APIs.
- Rename the [`RPCClient`](https://github.com/btcsuite/btcwallet/blob/93c858b2ad6390f574f73dc05d3c610cb78a8efc/chain/btcd.go#L27) struct to `BtcdClient` and change the `NewRPCClient` method to `NewBtcdClient`.

### 2️⃣ [lightningnetwork/lnd](https://github.com/lightningnetwork/lnd) Repo
- Modify the [`backendSupportsTaproot`](https://github.com/lightningnetwork/lnd/blob/572784a6a1f269762629c43a0ddd73e05352d663/chainreg/taproot_check.go#L11) function to depend on the `chain.Interface` abstraction instead of the concrete `rpcclient.Client` type.
- In the [`BitcoindEstimator`](https://github.com/lightningnetwork/lnd/blob/572784a6a1f269762629c43a0ddd73e05352d663/lnwallet/chainfee/estimator.go#L338) struct, use `*chain.BitcoindClient` instead of `*rpcclient.Client`.
- In the [`BtcdEstimator`](https://github.com/lightningnetwork/lnd/blob/572784a6a1f269762629c43a0ddd73e05352d663/lnwallet/chainfee/estimator.go#L144) struct, use `*chain.BtcdClient` instead of `*rpcclient.Client`.
- In the [`BtcdFilteredChainView`](https://github.com/lightningnetwork/lnd/blob/572784a6a1f269762629c43a0ddd73e05352d663/routing/chainview/btcd.go#L21) struct, use `*chain.BtcdClient` instead of `*rpcclient.Client`.

What do you think? 🙏

**Comment by yyforyongyu** [comment]
@mohamedawnallah yeah would be nice to see some draft PRs - I have tried locally and realized there are many places need to be changed so that's why the issue was created to track it. In the end, it's be nice to have the chain backend initialized once and once only here,
https://github.com/lightningnetwork/lnd/blob/baa3b0dea33a044e885821ff1557dba25ea7984b/server.go#L2111-L2118
